src/emitter/emitter.ts

```diff
@@ -63,15 +63,15 @@ export class EventEmitter<E extends object> implements IEventEmitter<E> {
   public createEmitGroup(): IEmitGroup<E> {
     const notifier = this;
 
-    const eventQueue: Array<[string, unknown[]]> = [];
+    const eventQueue: Array<[keyof E, unknown[]]> = [];
```

Comment:
```suggestion
    const eventQueue: Array<[keyof E, Parameters<E[key of E]]> = [];
```
Wouldn't this be slightly more correct?

Reply:
I think I tried that... the problem is that ends up creating an N * M type. Rather than the desired:
```
[ key1, params1 ] | [ key2, params2 ]
```
it creates a:
```
[ key1 | key2, params1 | params2 ]
```

which is actually wrong. (And also a combinatorial explosion.) And I was too dumb to see what the correct type is.

I _think_ it is something very very similar to the types for `on<E>` and `emit<E>` though... let me see if I can whip up a parameterized expression that gets it correct.

Reply:
Alrighty, I came up with:
```ts
    type EventQueueTuple = { [P in keyof E]: [P, Parameters<E[P]>] }[keyof E];

    const eventQueue: EventQueueTuple[] = [];
```

that lets us get rid of the `any`s and `unknown`s.